content/docs/plugin-communication.md

```diff
@@ -2669,3 +2672,17 @@ The general layout of the following notifications look like this
 	idFrom:		BufferID
 
 ---
+
+### Notification BufferID
+
+The BufferID received by a notification is not necessarily the _active_ buffer.  If you want to perform an action on the buffer that requires that it be active, you will have to activate that file first.  One reasonable sequence of events for doing some action on a specific BufferID in a callback is as follows:
+
+1. Store the BufferID of the active file (`keepBufferID`)
+    - [NPPM_GETCURRENTBUFFERID](#2084nppm_getcurrentbufferid) can be used to populate `keepBufferID`
+2. If the notification BufferID is not the `keepBufferID`, then activate the BufferID.
+	- [NPPM_GETPOSFROMBUFFERID](#2081nppm_getposfrombufferid)`(BufferID)` extracts a value p for the notification BufferID.
+	- Set `v = p >> 30`; set `i = p & 0x3FFFFFFF`.
+	- [NPPM_ACTIVATEDOC](#2052nppm_activatedoc)`(v,i)` will activate the notification BufferID
+3. Perform your actions on the notification BufferID, which is now active.
+4. Set the `keepBufferID` as the active file.
+    - use the same messages as in step 2, but starting with `NPPM_GETPOSFROMBUFFERID(keepBufferID)`

```

Comment:
* Instead of `Set v = p >> 30; set i = p & 0x3FFFFFFF` can we have the better `Set view = p >> 30; set docIndex = p & 0x3FFFFFFF` -- and then later the corresponding `(view, docIndex)` ?
* In step 1, how about adding at the end: `e.g. set keepBufferID = [NPPM_GETCURRENTBUFFERID](...)()